lib/src/js/polymer_base_dart.html

```diff
@@ -91,10 +105,32 @@
         // If you call notifyPath from dart we need to do manual modifications
         // on any JsArray or JsObject instances.
         var parts = this._getPathParts(path);
-        if (parts.length > 1) {
-          var model = this.get(parts.splice(0, parts.length - 1));
-          if (model[parts[0]] != value) model[parts[0]] = value;
+        // if (parts.length > 1) {
+
+        var thisArg =  this;
+        for (var i=0;i<parts.length-1;i++) {
```

Comment:
In the end maybe notifyPath will only fire the change event and nothing else. It will not change anything. So it is not a problem if it is called with a "wrong" path for normal users. But if you hook into it as you had to do then there are problems. Can it be a polymer bug then?